src/test/java/pages/HomePage.java

```diff
@@ -4,33 +4,24 @@
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
 import org.openqa.selenium.support.PageFactory;
+import pageComponents.HeaderGeneralPageComponent;
 
 public class HomePage {
-    public HomePage(WebDriver driver)   {
-        PageFactory.initElements(driver, this);
-    }
-
-    @FindBy(xpath = "//a[@href='/log-in']")
-    private WebElement signInBtn;
-
-    @FindBy(xpath = "//a[@href='/sign-up']")
-    private WebElement signUpBtn;
 
-    @FindBy(xpath = "//a[contains(.,'Restaurants List')]")
-    private WebElement restaurantsListBtn;
+    private final WebDriver driver;
 
     @FindBy(xpath = "//a[contains(.,'View All')]")
     private WebElement viewAllTag;
 
-    public void clickSignInBtn() {
-        signInBtn.click();
-    }
-    public void clickSignUpBtn() {
-        signUpBtn.click();
+    public HomePage(WebDriver driver)   {
+        this.driver = driver;
+        PageFactory.initElements(this.driver, this);
     }
-    public void clickRestaurantsListBtn() {
-        restaurantsListBtn.click();
+
+    public HeaderGeneralPageComponent getHeaderGeneralPageComponent() {
+        return new HeaderGeneralPageComponent(driver);
     }
```

Comment:
We can add initialization to the constructor

Reply:
Also don't sure if we need this method at all

Reply:
I'm not sure we have to initialize it in the constructor. I think there is no reason to use eager initialization.